src/unpoly/form.js

```diff
@@ -184,13 +184,18 @@ up.form = (function() {
   */
   function submittingButton(form) {
     const selector = submitButtonSelector()
-    const focusedElement = up.viewport.focusedElementWithin(form)
-    if (focusedElement && focusedElement.matches(selector)) {
-      return focusedElement
-    } else {
-      // If no button is focused, we assume the first button in the form.
-      return e.get(form, selector)
+    const focusedElement = document.activeElement
+
+    // Use focused element only if it is a part of the submitted form.
+    if (focusedElement && focusedElement.form === form) {
+      // It must match the submit button selector as well.
+      if (focusedElement.matches(selector)) {
```

Comment:
We can merge these two `if` statements into a single condition.

Reply:
> While MacOS does not focus buttons by default (which we rely on through our use of `document.activeElement`) we already address this by [explicitly focusing submit buttons when clicked](https://github.com/unpoly/unpoly/blob/ff7e379ad1b46c41b39a2c63ed03bfb26cce101e/src/unpoly/form.js#L609-L619).

This might sound weird, but it doesn't seem to work for me, but only in some cases and I can't seem to pinpoint where is the difference. Doing another explicit `.focus()` on those seems to fix the issue for me. The template I am converting to unpoly is pretty convoluted, though, so it might just be a local issue.